src/Compilers/CSharp/Portable/Symbols/Symbol_Attributes.cs

```diff
@@ -389,17 +389,17 @@ internal virtual void PostDecodeWellKnownAttributes(ImmutableArray<CSharpAttribu
 
             Debug.Assert(!earlyDecodingOnly);
 
-            this.PostDecodeWellKnownAttributes(boundAttributes, attributesToBind, diagnostics, symbolPart, wellKnownAttributeData);
-
-            removeObsoleteDiagnosticsForForwardedTypes(boundAttributes, attributesToBind, ref diagnostics);
-            Debug.Assert(diagnostics.DiagnosticBag is not null);
-
             // Store attributes into the bag.
             bool lazyAttributesStoredOnThisThread = false;
             if (lazyCustomAttributesBag.SetAttributes(boundAttributes))
             {
                 if (attributeMatchesOpt is null)
                 {
+                    this.PostDecodeWellKnownAttributes(boundAttributes, attributesToBind, diagnostics, symbolPart, wellKnownAttributeData);
```

Comment:
Why is it ok to move this from being unconditionally executed to only being executed if there's no matching filter?

Reply:
> Why is it ok to move this from being unconditionally executed to only being executed if there's no matching filter?

The only allowed side effect of this call is error reporting into ```diagnostics``` (from doc comment: "This method should not have any side effects on the symbol, i.e. it SHOULD NOT change the symbol state."), and this code path is the only code path that preserves the ```diagnostics```.

Reply:
Thanks for the explanation.